### DISCUSSION TITLE
How to apply multiple effects in composite?

### DISCUSSION THREAD

**Discussion by Georodin** [ImageMagick/ImageMagick#3539] [Help]
Hey guys,
Im currently struggling with I guess a simple composite. Im using the cli version only.
On my vps I have cronjob running wich prints every minute a timestamp onto a random picture - now the problem is that I need it in banner size.

Picture A - 1920x1080 random gaming background
Picture B - 1920x1080 current time in the lower left
Picture Output - 1920x1080 random gaming background /w current time
`composite PicB PicA PicOut` did the trick so far

but I want the output picture to be the following
1. merge PicA and PicB like always
2. scale the merge down to height 540
3. add as new background a blurred PicA unscaled
4. get the result in 1920x540
 
How could the call for this function look like? I tried to find similar examples but I failed. 

If you want examples Image examples I can provide sum. 
Thank you in advance and also thank you for this nice lightweight cli software!

**Comment by snibgo** [comment]
What version of IM, on what platform?

You use the "composite" program. I suggest _not_ using that. Instead, use "magick" for IM v7, or "convert" for IM v6. With either of these, include the "-composite" operation. The inputs to "-composite" should be in the opposite order to your command.

The command would be something like:
```
magick PicA PicB -composite -resize "1920x540!" PicA -blur 0x5 +swap -composite PicOut
```
This isn't efficient as it reads PicA twice. You want to "scale" an image, but I have assumed you want "-resize". I assume the merged image has transparency, otherwise compositing that over a blurred PicA won't do anything.

Please show sample input images.

**Reply by Georodin** [reply]
```
Version: ImageMagick 6.9.10-86 Q16 x86_64 2020-01-13 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5)
Delegates (built-in): bzlib cairo fftw fontconfig freetype gslib gvc jbig jng jp2 jpeg lcms ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zlib
```
On a CentOS 8 Redhat Linux shell

Here is what I like to produce:
[https://i.imgur.com/rGTa8wA.png](https://i.imgur.com/rGTa8wA.png)

Here is what I currently produce:
[https://i.imgur.com/tuvgxFz.jpg](https://i.imgur.com/tuvgxFz.jpg)

If you like me to provide additional information just ask.

**Comment by snibgo** [comment]
Please show sample input images.

**Comment by Georodin** [comment]
![6](https://user-images.githubusercontent.com/45157241/114582572-00056580-9c81-11eb-8d0b-3570058ba580.jpeg)
![0000](https://user-images.githubusercontent.com/45157241/114582786-38a53f00-9c81-11eb-974b-b64a98eda117.png)

**Comment by snibgo** [comment]
bash script:
```
convert \
  114582572-00056580-9c81-11eb-8d0b-3570058ba580.jpeg \
  \( +clone -blur 0x20 -write mpr:BLR +delete \) \
  114582786-38a53f00-9c81-11eb-974b-b64a98eda117.png \
  -composite \
  -resize x540 \
  mpr:BLR \
  +swap \
  -gravity Center \
  -composite \
  -crop x540+0+0 \
  out.jpg
```
![out](https://user-images.githubusercontent.com/33812368/114589605-bbc59580-9c7f-11eb-82c6-35d166acb2bc.jpg)

**Comment by GeeMack** [comment]
As **snibgo** showed in his example, you can use sets of parentheses in ImageMagick to keep various parts of a command isolated to do separate operations. Here is another example command that should produce the result in your example...

    convert PicA.jpg -write mpr:img0 -gravity center -extent 0x540 -blur 0x20 \
        \( mpr:img0 PicB.png -composite -resize x540 \) -composite output.png    

That produces this output image...

![output](https://user-images.githubusercontent.com/65475279/114609177-1ea93380-9c64-11eb-859d-8022c5dd6a32.png)
